### ISSUE TITLE
W45 Lib Team Call

### ISSUE THREAD

**Issue by nondiremanuel** [bitcoindevkit/.github#84]
:calendar: Tues November 7, 2023 12:00 UTC
:round_pushpin: Discord General Voice Channel

**Please add comments to this issue to suggest issues and PRs to discuss.**

## General Discussion Items

- [x] BDK 1.0 milestones and timeline (bitcoindevkit/bitcoindevkit.org#162: high level functional roadmap description blog post) -> upgrade rust-bitcoin to v0.31.1 (added an issue to alpha.3, but it could require another alpha for the miniscript part if it's not ready in time)
- [x] BDK Review Club: no review scheduled 
- [x] Any review begs? :pray: ci issue was fixed, PRs need to be rebased
- [x] Blockers? 🪨 bitcoindevkit/bdk#1182 blocking some PRs, needs review

#### Need Review:
- [x] bitcoindevkit/bdk#1178: review ongoing
- [x] bitcoindevkit/bdk#1028: review ongoing, with the caveat that after CI fixes are made on master branch it will need to be rebased -> still in progress after some comments
- [x] bitcoindevkit/bdk#1190: review ongoing -> probably ready to be merged
- [x] bitcoindevkit/bdk#1186: fixes #1127, needs review -> check if the comments were all addressed or not (and rebase due to ci fix)
- [x] bitcoindevkit/bdk#1117: waiting to be reviewed
- [x] bitcoindevkit/bdk#1109: changes approved by Lloyd. Depending on #1182, it could need rebase after ci fix
- [x] bitcoindevkit/bdk#1183: needs to wait LDK's https://github.com/lightningdevkit/rust-lightning/pull/2681 before merging -> it will be moved back to draft
- [x] bitcoindevkit/bdk#1161: waiting to be reviewed
- [x] bitcoindevkit/bdk#1176: seems to impact the API, so we assign it to alpha.3 for the moment. Waiting to be reviewed -> actually still in progress
- [x] bitcoindevkit/bdk#1156: it will need to be closed if we land to Nix ci
#### In progress:
- [x] bitcoindevkit/bdk#1167: WIP (daniela) -> probably it will be substituted by a new PR, close to be opened
- [x] bitcoindevkit/bdk#1042: WIP -> code already implemented. For now using KeychainKind
- [x] bitcoindevkit/bdk#1172: WIP, replaces 1106. -> moved to Needs Review
- [x] bitcoindevkit/bdk#1145: needs to be continued after discarding bitcoindevkit/bdk#1163 solution -> it doesn't seem to fix completely the related issue #1125. It could be merged but keeping the issue open, unlinking it
- [x] bitcoindevkit/bdk#1195: issue collecting different requirements for syncing. To be discussed. The work was started in  bitcoindevkit/bdk#1194 -> same PR or different one to implement Lloyd's suggestion
- [x] bitcoindevkit/bdk#1115: No update. Still on hold, needs some comment from evan and/or Lloyd. The old method has some problems but it's not clear the impact of it. Degree of complexity (and consequently milestone) needs to be discussed
- [x] bitcoindevkit/bdk#1128: No update. Another approach has to be tried -> will ping on discord
- [x] bitcoindevkit/bdk#1141: No update. Important to keep it since it affects the Wallet API. Not clear if it is close to end or not. Probably simplified by last rust-bitcoin version
- [x] bitcoindevkit/bdk#979: No update. Related to #1145, so included in alpha.3 -> waiting for #1171. Moved to beta since it is test related
#### To Do:
- [x] bitcoindevkit/bdk#1154: discord discussion on scan vs rescan, still ongoing. Probably included in super issue  bitcoindevkit/bdk#1195 (above) -> poll on discord
- [x] bitcoindevkit/bdk#1144: No update 
- [x] bitcoindevkit/bdk#1142: Bug to be investigated before alpha.3 and then evaluate the priority. mgravitt is asking for updates -> daniela will have a look on it (and we can tell max that it is ongoing)
- [x] bitcoindevkit/bdk#1180: new on transaction verification -> it shouldn't be very hard and lower priority for alpha.3
- [x] bitcoindevkit/bdk#1179: useful for android binding, possibly to be included in the next bdk_esplora release. Since it should be quick adding a flag, we added to alpha.3

### Milestone TBD:
- [x] bitcoindevkit/bdk#1072: WIP (evan): is it correct to keep it high priority being it labeled for 2.0 release?
- [x] bitcoindevkit/bdk#1188: needs review, but which milestone should it be assigned to? -> moved to alpha.3 since it should be small and makes things clearer
- [x] bitcoindevkit/bdk#1193 -> it could be required for alpha.3 since it could have breaking changes (ask also to evan)

## Other updates for future releases
- [x] bitcoindevkit/bdk#1165: ci to Nix; possibly in the future also for the ffi repo. Put in beta, if it comes earlier we can include it in alpha.3
- [x] bitcoindevkit/bdk#1177: adjustments needed for the release of rust-bitcoin v0.31.0-rc1: since not necessary for the 1.0 beta, put in 2.0 but if we can get it sooner it could be useful. update: rust-bitcoin release is out but still working on miniscript; fixes deprecation of is_coin_base (bitcoindevkit/bdk#1191)
- [x] bitcoindevkit/bdk#1171: in beta, if it comes earlier we can include it in alpha.3
- [x] bitcoindevkit/bdk#1158: needs review (it should be a quick one) -> very small change, moved to alpha.3

## Upcoming Events

- [x] [BTC Azores](https://btcazores.com/) Sept 23-24
- [x] [PlanB Lugano](https://planb.lugano.ch/?lang=it) Oct 20-21
- [x] Africa Bitcoin Conference (Accra) Dec 1-3 2023

🌅 *[Future Call Notes](https://github.com/bitcoindevkit/.github/issues?q=label%3ACall+is%3Aopen)*

📜 *[Past Call Notes](https://github.com/bitcoindevkit/.github/issues?q=label%3ACall+is%3Aclosed)*

[BDK Milestones]:https://github.com/bitcoindevkit/bdk/milestones
[BDK Core Staging Roadmap]:https://github.com/users/LLFourn/projects/2
[BDK 1.0]:https://github.com/orgs/bitcoindevkit/projects/14
[BDK Maintenance]: https://github.com/orgs/bitcoindevkit/projects/6
[BDK-CLI Roadmap]:https://github.com/orgs/bitcoindevkit/projects/7
[Documentation Roadmap]:https://github.com/orgs/bitcoindevkit/projects/9
[BDK Core Project]:https://github.com/users/LLFourn/projects/2/views/1
